api/server/routes_create_update.go

```diff
@@ -32,14 +33,36 @@ func (s *Server) handleRoutesPostPutPatch(c *gin.Context) {
 		handleErrorResponse(c, err)
 		return
 	}
-	if method != http.MethodPatch {
-		err = s.ensureApp(ctx, &wroute, method)
-		if err != nil {
-			handleErrorResponse(c, err)
-			return
-		}
+	appName := c.MustGet(api.App).(string)
+
+	appID, err := s.ensureApp(ctx, appName, &wroute, method)
+	if err != nil {
+		handleErrorResponse(c, err)
+		return
+	}
+
+	resp, err := s.ensureRoute(ctx, appID, method, &wroute)
```

Comment:
ordering of parameters should probably be consistent here between `ensureRoute` and `ensureApp` or appID and method will get flip flopped, it also seems like ensureApp probably doesn't need `wroute`

Reply:
fixed